### ISSUE TITLE
Full path completion

### ISSUE THREAD

**Issue by vincentisambart** [rust-lang/rust-analyzer#14325]
When I type for example `Inde` and ask for completion, I'd like to be able to complete to `std::ops::Index`.
Currently, if autoimport is enabled, I can choose `std::ops::Index` but rust-analyzer will add a `use` directive for it. And if autoimport is disabled, `std::ops::Index` is not in the completion list.

Something like `std::ops::Index` is something you are often going to reference only once in the whole file so adding a `use` for it doesn't make anything shorter. And I would like to not have to remember the full path, or do the completion with autoimport and then having to go cut the full path from the `use` statement that rust-analyzer added.

UI wise, when the user selects `std::ops::Index` and press not return but for example something like command+return that could complete the full path.

**Comment by Veykril** [comment]
Duplicate of https://github.com/rust-lang/rust-analyzer/issues/13953 (though your idea isn't listed there, if you could add that as comment that would be appreciated)
